ktor-features/ktor-velocity/jvm/src/io/ktor/velocity/Velocity.kt

```diff
@@ -29,10 +30,29 @@ public class VelocityContent(
     public val contentType: ContentType = ContentType.Text.Html.withCharset(Charsets.UTF_8)
 )
 
+internal class VelocityOutgoingContent(
+    val template: Template,
+    val model: Context,
+    etag: String?,
+    override val contentType: ContentType
+) : OutgoingContent.WriteChannelContent() {
+    override suspend fun writeTo(channel: ByteWriteChannel) {
+        channel.bufferedWriter(contentType.charset() ?: Charsets.UTF_8).use {
+            template.merge(model, it)
+        }
+    }
+
+    init {
+        if (etag != null) {
+            versions += EntityTagVersion(etag)
+        }
+    }
+}
+
 /**
  * Velocity ktor feature. Provides ability to respond with [VelocityContent] and [respondTemplate].
  */
-public class Velocity(private val engine: VelocityEngine) {
+public class Velocity private constructor(private val engine: VelocityEngine) {
```

Comment:
We can't make it private if it was public. All existing APIs should stay as they are.

Reply:
Fixed.